lib/sie/parser/build_entry.rb

```diff
@@ -19,38 +19,56 @@ def call
 
       def build_complete_entry
         entry = build_empty_entry
-        entry_type = first_token.entry_type
 
-        entry_type.each_with_index do |entry_type, i|
-          break if i >= tokens.size
+        attributes_with_tokens.each do |attr, *attr_tokens|
+          label = attr.is_a?(Hash) ? attr[:name] : attr
```

Comment:
I'd do `attr.fetch(:name)`, since we always expect name to be there when it's a Hash (right?). Otherwise the nil can cause some error far from the place where it was introduced.